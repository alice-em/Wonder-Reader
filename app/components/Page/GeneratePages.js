import React, { memo } from 'react';

import Context from '../../context/ContextFactory';
import Page from './Page';

const generateTotalSize = encodedPages => encodedPages.reduce((a, { width }) => a + width, 0);

const GeneratePages = () => (
  <Context.Consumer>
    {({ state: { encodedPages }}) => (
      encodedPages.map(({ key, page, width }) => (
        <Page
          key={key}
          id={`page${key}`}
          width={(width / generateTotalSize(encodedPages)) * 100}
          alt="comic page"
          src={page}
        />
      ))
    )}
  </Context.Consumer>
);

export { GeneratePages };
export default memo(GeneratePages);
