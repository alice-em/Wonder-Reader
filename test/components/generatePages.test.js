import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import React from 'react';

import GeneratePages, {
  generatePage,
  generateTotalSize,
  generateTotalSizeBeta
} from '../../app/components/generatePages';

Enzyme.configure({ adapter: new Adapter() });

const GeneratePage = generatePage(1000);
const props = {
  pages: [
    {
      key: 0,
      page: 'abc.png',
      width: 100
    },
    {
      key: 1,
      page: 'deb.png',
      width: 300
    }
  ]
};
const sampleData = [1, 2, 3, 4, 5].map(n => ({ width: n }));

describe('generatePages', () => {
  it('should render', () => {
    const wrapper = shallow(<GeneratePages pages={props.pages} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('generatePage', () => {
    const wrapper = shallow(<GeneratePage {...props[0]} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('generateTotalSize', () => {
    expect(generateTotalSize(sampleData)).toBe(15);
  });

  it('generateTotalSizeBeta', () => {
    expect(generateTotalSizeBeta(sampleData)).toBe(15);
  });
});
