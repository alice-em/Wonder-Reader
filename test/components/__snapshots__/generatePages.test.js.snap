// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generatePages generatePage 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "alt": "comic page",
      "src": undefined,
      "width": NaN,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "alt": "comic page",
        "src": undefined,
        "width": NaN,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`generatePages should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <generatePages
    pages={
      Array [
        Object {
          "key": 0,
          "page": "abc.png",
          "width": 100,
        },
        Object {
          "key": 1,
          "page": "deb.png",
          "width": 300,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [
    <Page
      alt="comic page"
      src="abc.png"
      width={25}
    />,
    <Page
      alt="comic page"
      src="deb.png"
      width={75}
    />,
  ],
  Symbol(enzyme.__nodes__): Array [
    Array [
      <Page
        alt="comic page"
        src="abc.png"
        width={25}
      />,
      <Page
        alt="comic page"
        src="deb.png"
        width={75}
      />,
    ],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
